lib/CombinedCancellationToken.php

```diff
@@ -19,7 +19,7 @@ final class CombinedCancellationToken implements CancellationToken
     public function __construct(CancellationToken ...$tokens)
     {
         foreach ($tokens as $token) {
-            $id = $token->subscribe(function (CancelledException $exception) {
+            $id = $token->subscribe(static function (CancelledException $exception) {
```

Comment:
How is that supposed to work out? `static` strips the `$this` reference, but the closure is referencing `$this`.